.github/workflows/earthly-build.yaml

```diff
@@ -17,57 +17,92 @@ jobs:
         run: |
           sudo wget https://github.com/earthly/earthly/releases/latest/download/earthly-linux-amd64 -O /usr/local/bin/earthly
           sudo chmod 755 /usr/local/bin/earthly
-      - name: Build Space ROS
+      - name: Build Test Space ROS
         id: build
         run: |
-          earthly --ci --output +sources
-          earthly --ci --output +build-testing
-          echo "archivename=$(basename log/build_results_archives/build_results_*.tar.bz2)" >> $GITHUB_ENV
+          earthly --ci --output +build-test --VCS_REF="$(git rev-parse HEAD)"
       - name: Upload build results archive
         uses: actions/upload-artifact@v3
         with:
-          name: ${{env.archivename}}
-          path: log/build_results_archives/${{env.archivename}}
+          name: build_results.tar.bz2
+          path: build_test_results.tar.bz2
           if-no-files-found: error
-  space-ros-image:
+  space-ros-main-image:
     runs-on: ubuntu-latest
+    env:
+      IMAGE_TAG: "latest"
```

Comment:
We need to be careful with this, `latest` should only be for tagged builds, not `main`, as for us `latest` == the latest stable tagged version.

Given the build push step in https://github.com/space-ros/space-ros/pull/186/files#diff-2705f6016a64189abd14041e2f3e8dbb829553295d9088e4eb42eccb5e61bf47R409, maybe for this the simplest thing to just add one more build step? So set this to `${{ github.ref_name }}`, and then after the main push we just add another step:

```
      # For tagged releases, mark it as the latest
       - name: Push spaceros images to Dockerhub
         if: ${{ github.ref_type == 'tag' }}
         run: |
           echo $DOCKER_HUB_TOKEN | docker login --username osrfbot --password-stdin
           earthly --ci --push +push-main-image --IMAGE_TAG="latest" --VCS_REF="$VCS_REF"
```

I'm not sure if we can add multiple tags like was in there before, so this adds a modicum of time for releases that won't affect other builds. What do you think?

Reply:
I had some confusion related to this. Currently, we have different tags to manage. I think it depends on the branching strategy we are about to make. 

To me `latest` should always refer to `main`. Any milestones when it is declared should go into their branch thus the image built on top of these branches should have the tag of the milestones. More information can be found in the discussion [here](https://github.com/space-ros/space-ros/discussions/140#discussioncomment-10276994).

How should we proceed?

Reply:
> To me latest should always refer to main

I tend to agree but am not sure what the consensus was coming out of that discussion. @Bckempa what do you think? IMO making `latest` == `main` will actually force users to use tags in their projects, which is better practice anyway. Also, we're going to be doing it in https://github.com/space-ros/docker/issues/197 anyway, so can we go ahead and make that change in this PR?

Reply:
I am trying to digest the information. So long story short, 

 - We will be aiming the `latest` tag to the recently released and stable version. This goes for both the `main` variant and the `dev` variant of the images. This stage gets into dockerhub tagged as the `latest` and the `<tag_name>`.
 - The GitHub registry will hold the image of the `main` variant to the latest commit of the `main` branch. This stage stays within GHCR for the organization to use with the latest commit. The tag name for this image can be simply `main` which will be used by CI across the organization.

It looks like calling the core variant of the image as the main variant is confusing at this point. Should I revert it to be called core?

Can you tell me if I had grasped correctly or am I missing something?